### PR TITLE
Added docs about cluster-autoscaler balanceSimilarNodeGroups

### PR DESCRIPTION
**Pull request by AverageMarcus**
### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [ ] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/main/CONTRIBUTING.md#front-matter) associated with any updated docs

Related release: https://github.com/giantswarm/cluster-autoscaler-app/releases/tag/v1.22.2-gs3

⚠️  Blocked on 17.0.0 release that'll include the relevant version of cluster-autoscaler

### CODE REVIEW

File: src/content/advanced/cluster-autoscaler/index.md [hunk 1]
```diff
@@ -126,6 +126,19 @@ data:
       skipNodesWithLocalStorage: "true"
 ```
 
+### Balance similar node groups
+
+> Added in release v1.22.2-gs3
```
Comment (whites11) [inline src/content/advanced/cluster-autoscaler/index.md:131]:
```suggestion
> Added in release v1.23.0-gs1
```

File: src/content/advanced/cluster-autoscaler/index.md [hunk 2]
```diff
@@ -126,6 +126,19 @@ data:
...
+
+The Cluster Autoscaler configuration by default doesn't differentiate between node groups when scaling. In case you want to enable this action, you need to set the following property to `"true"`.
```
Comment (whites11) [inline src/content/advanced/cluster-autoscaler/index.md:133]: I would also link to https://docs.giantswarm.io/app-platform/app-configuration/ for details about how to submit the custom config

Comment (AverageMarcus) [reply src/content/advanced/cluster-autoscaler/index.md:133]: It's already descibed on the page: https://docs.giantswarm.io/advanced/cluster-autoscaler/#how-to-set-configuration-options-using-the-user-values-configmap

**Review by whites11 — APPROVED** [review]

**Review by fiunchinho — APPROVED** [review]

**Comment by alex-dabija** [comment]
AWS v17.0.0 is out. I think the PR can be merged.

**Comment by marians** [comment]
Just noted that we use different versions of "Cluster Autoscaler", "cluster-autoscaler". Would be nice to settle for one version. FYI: We only use `Title Case` for some select terms which we consider important names. In this case we should not do it.

**Pull request merged**